src/NHibernate.Test/UtilityTest/CollectionHelperFixture.cs

```diff
@@ -180,6 +182,69 @@ public void SetsWithoutSameCountShouldBeInequal()
 			Assert.That(CollectionHelper.SetEquals(c2, c1), Is.False);
 		}
 
+		[Test]
+		public void MapsWithSameContentShouldHaveSameHashCode()
+		{
+			var d1 = new Dictionary<string, string> { { "1", "2" }, { "3", "4" } };
+			var d2 = new Dictionary<string, string> { { "1", "2" }, { "3", "4" } };
+			Assert.That(
+				CollectionHelper.GetHashCode(d1, KvpStringComparer.Instance),
+				Is.EqualTo(CollectionHelper.GetHashCode(d2, KvpStringComparer.Instance)));
+		}
+
+		// Failure of following tests is not an error from GetHashCode semantic viewpoint, but it causes it
```

Comment:
Better to replace asserts with warnings then

Reply:
The trouble with such warnings is that their failures will likely be unnoticed. A test that may just generate warnings without having any assert does no good in my opinion.

Even if the `GetHashCode` semantics allows collision, having collisions on simple and recurring cases is a bad thing and is worth a failing test case in my opinion.

Reply:
NUnit’s warnings mark tests as inconclusive.

Reply:
Yes but who checks inconclusive tests? If the build does not fail, a PR causing a regression on this point is very likely to not be rejected due to causing that regression, because it will likely goes unnoticed.